editoast/src/views/infra/mod.rs

```diff
@@ -222,21 +221,30 @@ async fn list(
         ..
     }): State<AppState>,
     Extension(auth): AuthenticationExt,
-    Query(pagination_params): Query<PaginationQueryParams<1000>>,
+    Query(pagination): Query<PaginationQueryParams<1000>>,
 ) -> Result<Json<InfraListResponse>> {
     let authorized = auth
         .check_roles([Role::OperationalStudies, Role::Stdcm].into())
-        .await
-        .map_err(AuthorizationError::AuthError)?;
+        .await?;
     if !authorized {
         return Err(AuthorizationError::Forbidden.into());
     }
 
-    let settings = pagination_params.into_selection_settings();
+    let conn = &mut db_pool.get().await?;
+    let settings = pagination.into_selection_settings();
 
-    let (infras, stats) = {
-        let conn = &mut db_pool.get().await?;
-        Infra::list_paginated(conn, settings).await?
+    let (infras, stats) = match auth.list_authorized_infra().await? {
+        Authorization::Granted(infra_ids) => {
+            Infra::list_paginated(
+                conn,
+                settings.filter(move || Infra::ID.eq_any(infra_ids.clone())),
+            )
+            .await?
+        }
+        Authorization::Bypassed => Infra::list_paginated(conn, settings).await?,
+        Authorization::Denied { reason: _ } => {
+            return Err(AuthorizationError::Unauthorized.into());
+        }
     };
```

Comment:
nit: the match could more precisely just return the settings, the list_paginated doesn't really need to change

Reply:
The match also has the Deny arm, that's why I did that, even if at this step we are sure that the user is connected and with the good permissions.
Otherwise I need a default arms where I give access to  the full access to the infra list.

Reply:
Default is implemented for `SelectionSettings`. I meant something in the lines of:

```suggestion
	let settings = match auth.list_authorized_infra().await? {
        Authorization::Granted(infra_ids) => {
            settings.filter(move || Infra::ID.eq_any(infra_ids.clone()))
        }
        Authorization::Bypassed => Default::default(),
        Authorization::Denied { reason } => {
            unreachable!("user is authenticated at this point: {reason}")
        }
    };
    let (infras, stats) = Infra::list_paginated(conn, settings).await?;
```

There is an `unreachable!` macro for this use case. It panics if it we reach it. It helps highlight the bug instead of silently ignoring it.

Reply:
I was not aware of this macro, thanks. 
Will change that